src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -93,8 +95,15 @@ private void InitializeColors()
             CodeAnalysisColors.AccentBarColorKey = EnvironmentColors.FileTabInactiveDocumentBorderEdgeBrushKey;
         }
 
-        protected override void LoadComponentsInUIContext(CancellationToken cancellationToken)
+        protected override async Task LoadComponentsInUIContextAsync(CancellationToken cancellationToken)
         {
+            await GetServiceAsync(typeof(SVsTaskStatusCenterService));
+            await GetServiceAsync(typeof(SVsErrorList));
+            await GetServiceAsync(typeof(SVsSolution));
+            await GetServiceAsync(typeof(SVsShell));
+            await GetServiceAsync(typeof(SVsRunningDocumentTable));
+            await GetServiceAsync(typeof(SVsTextManager));
```

Comment:
are these not cancellable?  
no configure-awaiting?  can you please put .ConfigureAwait(true) to indicate intent?

Reply:
> This is a JTF aware class

How does one know that? I ask honestly since i'm trying to figure out the how to assess code during reviews.  I can accept that JTF stuff odesn't need this.  But i'm curious how one tells the difference when looking at something like this.  Thanks!

Reply:
> How does one know that?

We don't have a great way to tell at the moment. So far only the package initialization code was migrated to the JTF form at the time we switched to `AsyncPackage`.

Reply:
Ok.  And the rule is: in the package code *itself* it should not configure-await?  Does that also apply to any async methods in *other* services that it calls into?

Reply:
And, for now, should i just consider this in AsyncPackage, and AbstsractLanguageService and those related classes?  or is this just AsyncPackage?

Reply:
> in the package code itself it should not configure-await?

It can, but it's not a necessity and can hurt performance by increasing the number of context switches. This especially applies to the initialization code where things tend to need the main thread.

> Does that also apply to any async methods in other services that it calls into?

No, callees can do whatever they want as long as they locally adhere to the [threading rules](https://github.com/Microsoft/vs-threading/blob/master/doc/threading_rules.md).

> And, for now, should i just consider this in AsyncPackage, and AbstsractLanguageService and those related classes? or is this just AsyncPackage?

Everything derived from `AsyncPackage`, including but not limited to types derived from `AbstractPackage`.

Reply:
> It can, but it's not a necessity and can hurt performance by increasing the number of context switches.

Note: to be clear, i wasn't asking for ConfigureAwait(false).  I was specifically just asking about *explicit* ConfigureAwaits.  

The reason i like explicit (regardless of if its true/false) is that it makes it clear that the author thought about this and asked for the right semantics.  So even if `.ConfigureAwait(true)` is technically unnecessary, it helps make it clear the intent of the code without someone having to go:  "was that an accidental or intentional omission?".